terraform/gcp/modules/icl-cluster/main.tf

```
@@ -1,15 +1,102 @@
 resource "google_container_cluster" "cluster" {
   name = var.cluster_name
-  # count per zone
+  min_master_version = var.node_version
   initial_node_count = 1
+  remove_default_node_pool = true
+  deletion_protection = false
+}
+
+resource "google_container_node_pool" "pool" {
+  count = var.gpu_enabled ? 0 : 1
+  name       = "pool"
+  cluster    = google_container_cluster.cluster.name
+  node_count = 1
```

Let's use format that is consistent with the rest of the file (do not align equal signs)
```suggestion
  name = "pool"
  cluster = google_container_cluster.cluster.name
  node_count = 1
```